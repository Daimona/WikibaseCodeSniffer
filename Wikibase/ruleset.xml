<?xml version="1.0"?>
<ruleset name="Wikibase">
	<!-- See https://github.com/wikimedia/mediawiki-tools-codesniffer/blob/master/MediaWiki/ruleset.xml -->
	<rule ref="vendor/mediawiki/mediawiki-codesniffer/MediaWiki">
		<!-- We disagree with enforcing this as a hard rule -->
		<exclude name="MediaWiki.WhiteSpace.SpaceBeforeSingleLineComment" />
	</rule>

	<!-- NOTE: We purposely decided against all Generic.CodeAnalysis.… sniffs, because they all have
		possible exceptions, and are not meant to block patches from being merged. -->

	<rule ref="Generic.ControlStructures" />

	<rule ref="Generic.Files.InlineHTML" />
	<rule ref="Generic.Files.LineLength">
		<properties>
			<!-- The inherited "lineLimit" from the base MediaWiki ruleset is 100. -->
			<property name="absoluteLineLimit" value="140" />
		</properties>
		<!-- Exclude auto-generated files from the Translate extension, see magic-export.php. -->
		<exclude-pattern>\.i18n\.magic\.php</exclude-pattern>
	</rule>
	<rule ref="Generic.Files.OneClassPerFile" />
	<rule ref="Generic.Files.OneInterfacePerFile" />
	<rule ref="Generic.Files.OneTraitPerFile" />

	<!-- NOTE: Never add the Generic.Metrics.… sniffs to this generic rule set, because these are
		not about "code style", and the exact limits are highly disputable. -->

	<rule ref="Generic.PHP.CharacterBeforePHPOpeningTag" />

	<rule ref="PSR1">
		<exclude name="PSR1.Files.SideEffects" />
	</rule>
	<rule ref="PSR1.Methods.CamelCapsMethodName.NotCamelCaps">
		<!-- Exclude test methods like "testGivenInvalidInput_methodThrowsException". -->
		<exclude-pattern>tests*Test*\.php</exclude-pattern>
	</rule>

	<rule ref="PSR2.Files" />

	<!-- NOTE: We can not use the Squiz.Arrays.ArrayBracketSpacing sniff because it conflicts with
		the MediaWiki style that encourages to use spaces inside brackets. -->
	<rule ref="Squiz.Classes.DuplicateProperty" />
	<rule ref="Squiz.Classes.SelfMemberReference" />
	<rule ref="Squiz.ControlStructures.ControlSignature" />
	<rule ref="Squiz.Functions.FunctionDuplicateArgument" />
	<rule ref="Squiz.Functions.GlobalFunction" />
	<rule ref="Squiz.Scope" />

	<!-- TODO: Do we want this to be an enforced rule for all our code?
	<rule ref="Squiz.Strings.DoubleQuoteUsage">
		<exclude name="Squiz.Strings.DoubleQuoteUsage.ContainsVar" />
	</rule> -->

	<rule ref="Squiz.WhiteSpace.FunctionSpacing">
		<properties>
			<property name="spacing" value="1" />
		</properties>
	</rule>
	<rule ref="Squiz.WhiteSpace.OperatorSpacing">
		<properties>
			<property name="ignoreNewlines" value="true" />
		</properties>
	</rule>

	<arg name="extensions" value="php" />
	<arg name="encoding" value="utf8" />
	<exclude-pattern type="relative">^vendor/</exclude-pattern>
</ruleset>
