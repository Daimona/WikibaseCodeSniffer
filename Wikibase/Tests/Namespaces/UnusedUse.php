<?php

use Sub\Unused;
use UsedButBadCapitalization;
use Wikibase\AnotherBadCapitalization;
use UnusedMain ;
use UsedMain ;
use AnotherUnusedMain;
use Sub\Used;
use Wikibase\Sub\UsedInAnArray;
use UsedAndCommented;
use MentionedInComment;
use AppearsToBeUnused as ButIsUsed;
use Sub\AnotherAlias as /* Comments should be skipped */WithAComment;
use Sub\AndAnotherAlias as /** @note PHPDocs should also be skipped */WithADocComment;
use UsedAfterParamName;
use UsedAfterVarName;
use UsedAfterPipe;
use UsedTrait;
use UsedExpectedException;
use UsedThrows;
use UsedReturn;

/**
 * @group Unused
 */
class Example implements UsedMain {

	use UsedTrait;

	/**
	 * @type Used
	 */
	private $prop;

	/**
	 * @var UsedAndCommented with a comment
	 */
	private $prop;

	/**
	 * @var UsedButBadCapiTaliZation
	 */
	private $prop;

	/**
	 * @expectedException UsedExpectedException
	 * @param int[]|UsedInAnArray[] $arg
	 * @param bool $arg MentionedInComment should not count as usage
	 * @param $arg UsedAfterParamName[]|null
	 * @throws UsedThrows
	 * @return UsedReturn
	 */
	private function process( array $arg, AnotherBadCapiTaliZation $arg ) {
		/* @var ButIsUsed $var */
		/* @var $var UsedAfterVarName[]|null */
		/* @var $var null|UsedAfterPipe */
		/* @var $var bool MentionedInComment should not count as usage */
		$withAComment = new WithAComment();
		$withADocComment = new WithADocComment();
	}

}
