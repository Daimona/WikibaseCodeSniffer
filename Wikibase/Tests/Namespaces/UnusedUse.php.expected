  3 | ERROR | [x] Unused import found
  6 | ERROR | [x] Unused import found
  8 | ERROR | [x] Unused import found
 12 | ERROR | [x] Unused import found
